# Mushrooms

Brendan Reed  
11/11/2025  

Attempts to correctly identify mushrooms as poisonous or non-poisonous using classification.  
[Link to instructions](https://github.com/denisecase/ml-04/blob/main/CLASSIFICATION_PROJECT.md)

In [2]:
# Imports
from ucimlrepo import fetch_ucirepo
import numpy as np
import pandas as pd
import seaborn as sns


## Section 1. Import and Inspect the Data

### 1.1 Load the dataset and display the first 10 rows.

In [3]:
# fetch dataset
mushroom = fetch_ucirepo(id=73) 
  
# data (as pandas dataframes) 
data = mushroom.data.original
X = mushroom.data.features 
y = mushroom.data.targets

In [4]:
print(data.head(10))

  cap-shape cap-surface cap-color bruises odor gill-attachment gill-spacing  \
0         x           s         n       t    p               f            c   
1         x           s         y       t    a               f            c   
2         b           s         w       t    l               f            c   
3         x           y         w       t    p               f            c   
4         x           s         g       f    n               f            w   
5         x           y         y       t    a               f            c   
6         b           s         w       t    a               f            c   
7         b           y         w       t    l               f            c   
8         x           y         w       t    p               f            c   
9         b           s         y       t    a               f            c   

  gill-size gill-color stalk-shape  ... stalk-color-above-ring  \
0         n          k           e  ...                      w  

### 1.2 Check for missing values and display summary statistics.

In [5]:
print("info: ")
data.info()
print(f"describe: \n{data.describe()}")
print(f"sum of null values: \n{data.isnull().sum()}")
print(f"columns: \n{data.columns}")

info: 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8124 entries, 0 to 8123
Data columns (total 23 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   cap-shape                 8124 non-null   object
 1   cap-surface               8124 non-null   object
 2   cap-color                 8124 non-null   object
 3   bruises                   8124 non-null   object
 4   odor                      8124 non-null   object
 5   gill-attachment           8124 non-null   object
 6   gill-spacing              8124 non-null   object
 7   gill-size                 8124 non-null   object
 8   gill-color                8124 non-null   object
 9   stalk-shape               8124 non-null   object
 10  stalk-root                5644 non-null   object
 11  stalk-surface-above-ring  8124 non-null   object
 12  stalk-surface-below-ring  8124 non-null   object
 13  stalk-color-above-ring    8124 non-null   object
 14  stalk-color-below

Reflection 1: What do you notice about the dataset? Are there any data issues?  
Stalk root is missing 2480 instances out of 8124.

## Section 2. Data Exploration and Preparation

### 2.1 Explore data patterns and distributions

### 2.2 Handle missing values and clean data


### 2.3 Feature selection and engineering

Reflection 2: What patterns or anomalies do you see? Do any features stand out? What preprocessing steps were necessary to clean and improve the data? Did you create or modify any features to improve performance?

## Section 3. Feature Selection and Justification

### 3.1 Choose features and target

### 3.2 Define X and y


Reflection 3: Why did you choose these features? How might they impact predictions or accuracy?

### 4.1 Split the data

### 4.2 Train model

### 4.3 Evalulate performance

## Section 5. Improve the Model or Try Alternates

5.1 Train an alternative classifier (e.g., Decision Tree, Random Forest, Logistic Regression) OR adjust hyperparameters on the original model.

5.2 Compare performance of all models across the same performance metrics.

Reflection 5: Which model performed better? Why might one classifier be more effective in this specific case?



## Section 6. Final Thoughts & Insights

### 6.1 Summarize findings

### 6.2 Discuss challenges faced

### 6.3 If you had more time, what would you try next?

Reflection 6: What did you learn from this project?